### PR TITLE
cli: Support cluster name flags in all non-sql client commands

### LINKED ISSUE #40437: cli: `cockroach quit` doesn't work when cluster name is specified
When a cluster name is specified, `cockroach quit` requests do not succeed:

```
roland@roland-clustername-0001:~$ ./cockroach quit --certs-dir certs
E190903 19:05:25.860569 20 rpc/heartbeat.go:74  peer node expects cluster name "roland-clustername", use --cluster-name to configure
Error: Failed to connect to the node: initial connection heartbeat failed: cluster name check failed on ping response: peer node expects cluster name "roland-clustername", use --cluster-name to configure
Failed running "quit"
```

`cockroach quit` does not accept a `--cluster-name` argument: it just prints its usage if you supply that argument.

(Are there other non-SQL CLI commands that will need to be updated, or are they all deprecated now?)

### PR DESCRIPTION
**Pull request by a-robinson**
Unless I'm missing something, the cluster-name functionality is pretty
broken without the ability to use all the various client commands on
clusters that have a name. I was trying out some new features today and
couldn't figure out any supported way to do basic operations on a node
that was using a cluster-name. It seemed easy enough to fix, so here we
are.

I'm not at all tied to to including both the --cluster-name and
--disable-cluster-name-verification flags, but at first glance it seemed
useful to include both.

I didn't include any error cases in the tcl test because I didn't want
to have to figure out the correct tcl for dealing with the fact that the
commands hang in a seemingly infinite retry loop when the cluster-name
is missing/incorrect.

Fixes #40437

Release note (cli change): Client commands such as init and quit now
support the --cluster-name and --disable-cluster-name-verification flags
in order to support running them on clusters that have been configured
to use a cluster name. Previously it was impossible to run such
commands against nodes configured with the --cluster-name flag.

---

Hi @knz!

This seems like a strong cherry-pick candidate for a 20.1.x patch release (and maybe even a 19.2.x release) since the `--cluster-name` flag is a basically a trap for anyone trying to use it without this fix, but I'd also wager that it's not considered particularly urgent for 20.1.0 given that @rolandcrosby's ticket has gone unfixed since last September.

### CODE REVIEW

**Comment by knz** [comment]
Thanks Alex! Happy to hear from you :-) 

The functionality was intending to have clients call GRPCUnvalidatedDial which disables both cluster ID and name check. I am surprised this didn't work / wasn't tested. I'll review your PR shortly.

**Review by knz — APPROVED** [review]
Ok I understand what's going on and actually yes this was intended to work the way you fix it. Sorry this part of the functionality slipped under the radar.

:lgtm: modulo the default fix

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48000#-:-M5fu1ch8x-WZZb2_hvo:b7simj3)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @a-robinson)

___
*[pkg/cli/flags.go, line 625 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48000#-M5ftqXV26HlIJ0VPR8U:-M5ftqXW-VNBdfgUEViS:b-z8kqv8) ([raw file](https://github.com/cockroachdb/cockroach/blob/4eb5a684db4a5a6533c114cb97b6ed830bc7cd37/pkg/cli/flags.go#L625)):*
> ```Go
> 		VarFlag(f, urlParser{cmd, &cliCtx, true /* strictSSL */}, cliflags.URL)
> 		VarFlag(f, clusterNameSetter{&baseCfg.ClusterName}, cliflags.ClusterName)
> 		BoolFlag(f, &baseCfg.DisableClusterNameVerification, cliflags.DisableClusterNameVerification, false)
> ```

try this:
```go
BoolFlag(f, &baseCfg.DisableClusterNameVerification, cliflags.DisableClusterNameVerification, baseCfg.DisableClusterNameVerification)
```

Then initialize `baseCfg.DisableClusterNameVerification` in `context.go`.

**Comment by knz** [comment]
> This seems like a strong cherry-pick candidate for a 20.1.x patch release (and maybe even a 19.2.x release)

Fully agreed. Feel free to issue the backport PRs (but I can also do it if you prefer)

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48000)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @a-robinson)

___
*[pkg/cli/flags.go, line 625 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48000#-M5ftqXV26HlIJ0VPR8U:-M5fuSN1CTV2dBaYTcXY:b359u99) ([raw file](https://github.com/cockroachdb/cockroach/blob/4eb5a684db4a5a6533c114cb97b6ed830bc7cd37/pkg/cli/flags.go#L625)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

try this:
```go
BoolFlag(f, &baseCfg.DisableClusterNameVerification, cliflags.DisableClusterNameVerification, baseCfg.DisableClusterNameVerification)
```

Then initialize `baseCfg.DisableClusterNameVerification` in `context.go`.
</blockquote></details>

oh actually the default is already set in `baseCfg.InitiDefaults()` so no need to change anything in context.go. But still change the last argument to `BoolFlag`, we don't want defaults expressed in multiple places.

**Review by a-robinson — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48000)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @knz)

___
*[pkg/cli/flags.go, line 625 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48000#-M5ftqXV26HlIJ0VPR8U:-M5gXjXq4weU4gz4FdC7:bfoirtm) ([raw file](https://github.com/cockroachdb/cockroach/blob/4eb5a684db4a5a6533c114cb97b6ed830bc7cd37/pkg/cli/flags.go#L625)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

oh actually the default is already set in `baseCfg.InitiDefaults()` so no need to change anything in context.go. But still change the last argument to `BoolFlag`, we don't want defaults expressed in multiple places.
</blockquote></details>

No problem, done. I changed the existing `start`/`start-single-node` flag to use it as well.

**Comment by a-robinson** [comment]
> Fully agreed. Feel free to issue the backport PRs (but I can also do it if you prefer)

https://github.com/cockroachdb/cockroach/pull/48016 and https://github.com/cockroachdb/cockroach/pull/48017, although I had to make them manually since `go get`ting the backport tool was failing on a dependency version issue that I didn't want to look into, so sorry about the cockroachdb/release team not being tagged properly -- github wouldn't let me.

**Pull request merged**
